.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,146 @@
+# pip3 install ruamel.yaml.cmd
+# yaml merge-expand pull_request.yml exp.yml &&
+#    diff -w -u pull_request.yml exp.yml
+#
+# github.com also has a powerful web editor that can be used without
+# committing.
+
+name: Pull Requests
+
+# yamllint disable-line rule:truthy
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Some jobs may not need submodules but for now our CMakeLists.txt
+# systemically downloads them anyway when missing at build time. Easier
+# and cleaner to clone everything at once.
+
+jobs:
+
+  doxygen:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: apt get doxygen graphviz
+        run: sudo apt-get -y install ninja-build doxygen graphviz
+
+      - name: list all warnings, warnings are not failures
+        run: cmake -GNinja -S doc -B docbuild && ninja -C docbuild -v doc
+
+      # Build again (it's very quick) so warnings don't go unnoticed
+      - name: fail and stop on first warning
+        run: printf 'WARN_AS_ERROR = YES\n' >> doc/sof.doxygen.in &&
+          ninja -C docbuild -v doc
+
+
+  testbench:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+        with: {fetch-depth: 0, submodules: recursive}
+
+      - name: docker
+        run: docker pull thesofproject/sof && docker tag thesofproject/sof sof
+
+      # testbench needs some topologies
+      - name: build topologies
+        run: ./scripts/docker-run.sh ./scripts/build-tools.sh -t
+      - name: build testbench
+        run: ./scripts/docker-run.sh ./scripts/rebuild-testbench.sh
+      - name: test
+        run: ./scripts/host-testbench.sh
+
+
+  SOF-build-only:
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        platform: [sue jsl tgl]
+
+    steps:
+
+      - uses: actions/checkout@v2
+        with: {fetch-depth: 0, submodules: recursive}
+
+      - name: docker
+        run: docker pull thesofproject/sof && docker tag thesofproject/sof sof
+
+      - name: xtensa-build-all
+        env:
+          PLATFORM: ${{ matrix.platform }}
+        run: ./scripts/docker-run.sh
+          ./scripts/xtensa-build-all.sh -r ${PLATFORM}
+
+  # Warning: there is a fair amount of duplication between 'build-only'
+  # and 'qemu-boot' because github does not support YAML anchors as of Jan
+  # 2021.  Defining our new actions would be overkill. Another popular
+  # option is to generate this file from a source with YAML anchors
+  # before committing it; also deemed overkill for the current amount of
+  # duplication.
+
+  SOF-qemu-boot:
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        # Compiler-based grouping, see HOST= in xtensa-build-all.sh The
+        # only reason for grouping is to avoid the matrix swarming the
+        # user interface and burying other checks.
+        platform: [imx8 imx8x imx8m,
```

Comment:
This kind for matrix brings in too many GitHub status, could you try to simplify here

Reply:
I can group more, please make a suggestion.

Reply:
Is the current PR or grouping simpler for Travis/Docker quotas ? I don't mind the extra items.

Reply:
These are the same, per-toolchain groups than in Travis, however Travis results are condensed in a single line in Github.

Reply:
@marc-hb I think let's keep only  `gcc-build-only`  `qemu-boot` `codecheck` (include doxygen, yamllint and other code scan checkpatch), `testbench` would be better.

Duplicated status like `Pull Requests / SOF-qemu-boot (imx8 imx8x imx8m)` and `Pull Requests / SOF-qemu-boot (byt cht)` can be merged.

Reply:
I can very easily group (all) platforms for the same job because our build and qemu scripts support multiple platforms.

I don't think I can group  different jobs like doxygen + yamllint (not sure I would want that either) because I don't think github supports grouping jobs.

Reply:
I think it's good, lets merge and fine tune if work flow shows us where we can improve.

Reply:
Thanks @lgirdwood. @xiulipan can you submit yourself the grouping and naming changes you wanted? Will require only a few characters change now.

As grouping is implemented by `xtensa-build-all.sh` and NOT by github, note larger groups mean more _platforms not built at all_ if an early platform fails to build. Unless we add some new error collection logic in `xtensa-build-all.sh`